### PR TITLE
Render overdue appointments when appointments are loaded and overdue sections feature is enabled

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/8346/render-ui-for-all-overdue-sections-when-feature-flag-is-enabled

### CODE REVIEW

**Comment by msasikanth** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3889** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/3889" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/3889?utm_source=stack-comment).

**Comment by msasikanth** [comment]
I purposefully didn't add the hide case in ui renderer. When the feature is not enabled, we need to show the older appointment list which is a paging source that is triggered from view effects. We can discuss how this should look like and then make the change.

**Review by janhavisinghh — APPROVED** [review]

**Comment by msasikanth** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/3889?utm_source=stack-comment).


Job Reference: OnLo8O7RZS58UTIXGtOY

**Pull request merged**
